exam/models/exam.py

```diff
@@ -159,16 +159,17 @@ class ExamResult(models.Model):
     @api.multi
     @api.depends('result_ids')
     def _compute_total(self):
-        total = 0.0
-        if self.result_ids:
-            for line in self.result_ids:
-                obtain_marks = line.obtain_marks
-                if line.state == "re-evaluation":
-                    obtain_marks = line.marks_reeval
-                elif line.state == "re-access":
-                    obtain_marks = line.marks_access
-                total += obtain_marks
-            self.total = total
+        for rec in self:
+            total = 0.0
+            if rec.result_ids:
+                for line in rec.result_ids:
+                    obtain_marks = line.obtain_marks
+                    if line.state == "re-evaluation":
+                        obtain_marks = line.marks_reeval
+                    elif line.state == "re-access":
+                        obtain_marks = line.marks_access
+                    total += obtain_marks
+                rec.total = total
 
```

Comment:
shift the rec.total = total 4 spaces back, because if there are no result_ids, we will need to setup 0 as total.